### ISSUE TITLE
Printing with best fit results in job rejected.

### ISSUE THREAD

**Issue by TaiTair** [nomacs/nomacs#339]
**Describe the bug**
Our printer should be able to handle 1200x1200 dpi in black or 2400x600 dpi in color.
The jobs nomacs sends to our printer get rejected.

**To Reproduce**
When I go to print a big picture (4000x4000). I select landscape to make it fill the page. It shows the preview correctly. I can modify dpi to make it fit the page more or less.
If I leave the defaults, nomacs sends a 30mB job to the printer from a 1.65mB photo. The job gets rejected by the printer. Nothing is printed.
If I increase dpi to have white borders around the image it prints without issues. 
Note: It seems like this has something to do with the picture using the full page.

If I print a smaller picture (1280x1280). I select landscape or portrait, modify whatever and it prints no problem. The job sent to the printer was 15mB from a 500kB file.
I suspect that this picture printed because its aspect ratio could not fill the entire page.
The aspect ratio of the 4000x4000 could fill the entire page.

Note2: Pictures sent to the printer from Nomacs always take an unusual amount of time to spool up, 5-10min. I think this has to do with the job size nomacs sends to the printer.
If I print the same 1280x1280 photo from google chrome the job sent to the printer is 270kB.
There is clearly something wrong with the size of jobs nomacs sends to printers.

tl;dr prints small pictures but not big pictures, jobs sent to printer way too big.

**Expected behavior**
When I hit print I expect it to start printing within the next 10 seconds.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 3.12.1.4206

**Comment by TaiTair** [comment]
Closing this because the issue is related to the Windows 10 print spooler.

https://support.microsoft.com/en-ca/help/919543/the-size-of-the-emf-spool-file-may-become-very-large-when-you-print-a

**Comment by TaiTair** [comment]
Re-opening because the emf spool does not explain how google chrome can print the same image with a <1mB print job.

**Comment by diemmarkus** [comment]
thanks for digging into this - I will revisit the printing & see if we can reduce resolution. How did you analyze the job size?

**Comment by TaiTair** [comment]
A little bit more of an update: It seems the printer drivers are related to the issue. We were using the XPS drivers for our printer and those had a lot of issues. Switching to the PS3 driver helped a lot to complete jobs but the size of jobs is still too big. It went down from 300mB to 90mB for the same photo.

I checked the job size by printing a big photo and then checking the print queue. In the print queue it shows the size. This is on Windows 10. I'm fairly certain this would not happen on linux. I can double check on linux if that can help.

**Comment by TaiTair** [comment]
Still an issue to this day. Jobs sent from Nomacs are way to big. 1 Photo takes 5min to print on a laser printer.
